portfolio/src/main/webapp/script.js

```diff
@@ -74,5 +87,44 @@ function showImageAndDescription(prefix){
    imageContainer.innerHTML = '';
    imageContainer.appendChild(imgElement);
 }
+
+/**
+ * Fetches data from the server and adds it to the DOM.
+ */
+function getData() {
+  console.log('Fetching data...');
+
+  // The fetch() function returns a Promise because the request is asynchronous.
+  const responsePromise = fetch('/data');
+
+  // When the request is complete, pass the response into handleResponse().
+  responsePromise.then(handleResponse);
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addDataToDom().
+ */
+function handleResponse(response) {
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addQuoteToDom() function.
+  textPromise.then(addDataToDom);
+}
+
+/** Adds data to the DOM. */
+function addDataToDom(data) {
+  console.log('Adding text to dom: ' + data);
+
+  const dataContainer = document.getElementById('data-container');
+  dataContainer.innerText = data;
+}
```

Comment:
When using promises it seems common to chain them as smaller inline functions if you're just passing data around. When there is more complex work to do they are split into standalone functions like you have done.

eg (just for reference, there's nothing wrong with how you've done it :), especially if it makes it clearer to follow)
```
fetch('/data')
.then(response => response.text())
.then((data) => {
  const dataContainer = document.getElementById('data-container');
  dataContainer.innerText = data;
})
```